File: frontend/src/types/organization.ts
export interface Organization {
  id: string;
  user_id: string;
  name: string;
  industry: string | null;
  country: string | null;
  reporting_year: number | null;
  reduction_target_pct: number | null;
  base_year: number | null;
  base_year_emissions_tco2e: number | null;
  revenue_gbp_m: number | null;
  onboarding_complete: boolean;
  created_date: string;
}

export interface OrganizationCreatePayload {
  name: string;
  industry?: string;
  country?: string;
  reporting_year?: number;
  reduction_target_pct?: number;
  base_year?: number;
  revenue_gbp_m?: number;
  onboarding_complete?: boolean;
}

File: backend/app/models/organization.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Float, Integer, Boolean, DateTime, ForeignKey, CheckConstraint
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.config.database import Base


class Organization(Base):
    __tablename__ = "organizations"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    name = Column(String, nullable=False)
    industry = Column(String, nullable=True)
    country = Column(String, nullable=True)
    reporting_year = Column(Integer, nullable=True)
    reduction_target_pct = Column(Float, nullable=True)
    base_year = Column(Integer, nullable=True)
    base_year_emissions_tco2e = Column(Float, nullable=True)
    revenue_gbp_m = Column(Float, nullable=True)
    onboarding_complete = Column(Boolean, default=False)
    created_date = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    __table_args__ = (
        CheckConstraint(
            "industry IN ('manufacturing','construction','energy','chemicals','metals','logistics','other') OR industry IS NULL",
            name="check_industry"
        ),
    )

    user = relationship("User", back_populates="organisations_owned")
